zerver/lib/avatar_hash.py

```diff
@@ -15,25 +15,26 @@ def gravatar_hash(email: str) -> str:
     return hashlib.md5(email.lower().encode()).hexdigest()
 
 
-def user_avatar_hash(uid: str) -> str:
+def user_avatar_hash(uid: str, version: str) -> str:
     # WARNING: If this method is changed, you may need to do a migration
     # similar to zerver/migrations/0060_move_avatars_to_be_uid_based.py .
 
-    # The salt probably doesn't serve any purpose now.  In the past we
-    # used a hash of the email address, not the user ID, and we salted
-    # it in order to make the hashing scheme different from Gravatar's.
-    user_key = uid + settings.AVATAR_SALT
+    # The salt prevents unauthenticated clients from enumerating the
+    # avatars of all users.
+    user_key = uid + ":" + version + ":" + settings.AVATAR_SALT
     return hashlib.sha1(user_key.encode()).hexdigest()
 
 
-def user_avatar_path(user_profile: UserProfile) -> str:
+def user_avatar_path(user_profile: UserProfile, future: bool = False) -> str:
     # WARNING: If this method is changed, you may need to do a migration
     # similar to zerver/migrations/0060_move_avatars_to_be_uid_based.py .
-    return user_avatar_path_from_ids(user_profile.id, user_profile.realm_id)
+    return user_avatar_base_path_from_ids(
+        user_profile.id, user_profile.avatar_version + (1 if future else 0), user_profile.realm_id
+    )
```

Comment:
Probably worth adding a comment explaining the `future` detail, which I assume is that if we're about to write a new version, we want the next value.